packages/office-ui-fabric-react/src/components/TextField/TextField.scss

```diff
@@ -62,7 +62,8 @@
   font-size: $ms-font-size-m;
   color: $ms-color-neutralPrimary;
   height: 32px;
-  @include padding(6px, 12px, 7px, 12px);
+  line-height: 32px;
```

Comment:
Yes, just tested with Safari and it shows extra large cursor. The text is centered even without line-height, so we should be fine even without this. Thanks for pointing it out.